Configuration.props

```diff
@@ -6,26 +6,40 @@
   <Import
       Project="$(MSBuildThisFileDirectory)Configuration.Override.props"
       Condition="Exists('$(MSBuildThisFileDirectory)Configuration.Override.props')"
-  />
+      />
+
+  <!--
+      Note: this file *must* be imported *after* Configuration.Override.props
+  -->
+  <Import Project="$(MSBuildThisFileDirectory)Configuration.OperatingSystem.props" />
   <PropertyGroup>
     <HostOS Condition=" '$(HostOS)' == '' And '$(OS)' == 'Windows_NT' ">Windows</HostOS>
-    <HostOS Condition=" '$(HostOS)' == '' And '$(OS)' == 'Unix' And Exists ('/Applications') ">Darwin</HostOS>
-    <HostOS Condition=" '$(HostOS)' == '' And '$(OS)' == 'Unix' ">Linux</HostOS>
+    <HostCc Condition=" '$(HostCc)' == '' ">$(HostCc64)</HostCc>
+    <HostCxx Condition=" '$(HostCxx)' == '' ">$(HostCxx64)</HostCxx>
+    <HostCc Condition=" '$(HostCc)' == '' ">$(HostCc32)</HostCc>
+    <HostCxx Condition=" '$(HostCxx)' == '' ">$(HostCxx32)</HostCxx>
     <HostCc Condition=" '$(HostCc)' == '' ">cc</HostCc>
     <HostCxx Condition=" '$(HostCxx)' == '' ">c++</HostCxx>
+    <HostCcName Condition=" '$(HostCcName)' == '' ">cc</HostCcName>
+    <HostCxxName Condition=" '$(HostCxxName)' == '' ">c++</HostCxxName>
+    <NeedMxe Condition=" '$(HostOS)' == 'Darwin' ">true</NeedMxe>
+    <MakeConcurrency Condition=" '$(HostCpuCount)' != '' ">-j$(HostCpuCount)</MakeConcurrency>
```

Comment:
This should be:

    <MakeConcurrency Condition=" '$(MakeConcurrency)' == '' And '$(HostCpuCount)' != '' ">-j$(HostCpuCount)</MakeConcurrency>

This would allow explicitly overriding `$(MakeConcurrency)`, e.g. to diagnose one of those annoying `make -j4` build errors:

    xbuild /p:MakeConcurrency=1 ...